test/LoginStrategy.test.ts

```diff
@@ -0,0 +1,90 @@
+import assert from 'assert';
+import { config } from 'dotenv';
+import { Keys } from '@ew-did-registry/keys'
+import { ClaimsUser } from '@ew-did-registry/claims';
+import { ClaimData } from './claim-creation/ClaimData';
+import { ITokenPayload } from '../lib/LoginStrategy.types';
+import { ClaimsUserFactory } from './claim-creation/ClaimsUserFactory';
+import { LoginStrategy, LoginStrategyOptions } from '../lib/LoginStrategy';
+import { providers } from 'ethers';
+
+config();
+
+let firstClaimToken: string;
+let claimData: ClaimData;
+let payload: ITokenPayload;
+let claimsUser: ClaimsUser;
+let secondClaimToken: string;
+let otherclaimsUser: ClaimsUser;
+let loginProcess: LoginStrategy;
+let secondPayload: ITokenPayload;
+
+const initParams: LoginStrategyOptions = {
+    jwtSecret: process.env.PRIVATE_PEM,
+    jwtSignOptions: {
+      algorithm: 'RS256',
+    },
+    name: 'login',
+    rpcUrl: process.env.RPC_URL || 'https://volta-rpc.energyweb.org/',
+    cacheServerUrl: process.env.CACHE_SERVER_URL || 'https://identitycache-dev.energyweb.org/',
+    acceptedRoles: process.env.ACCEPTED_ROLES ? process.env.ACCEPTED_ROLES.split(',') : [],
+    privateKey: process.env.PRIVATE_KEY,
+  }
+
+const provider = new providers.JsonRpcProvider(initParams.rpcUrl)
+const keys = new Keys({
+    privateKey: process.env.PRIVATE_KEY,
+  });
+  const secondKeys = new Keys({
+    privateKey: process.env.PRIVATE_KEY_2,
+  });
+  const issuerDID = `did:ethr:${keys.getAddress()}`
+  const secondDID = `did:ethr:${secondKeys.getAddress()}`
+
+describe("Testing LoginStrategy", () => {
```

Comment:
A suggestion that I have is that you refactor the code that you have written into a separate class (similar to the current `ClaimVerifier`). In this way, you can write unit tests against the class that can test the various scenarios without needing the blockchain. In will also have the advantage of encapsulating your functions more directly. For example, you could have some class like:
```
AuthenticationTokenVerifier{
  public verify(token, didDocument) {
     ...
  }
  .... all of your private methods
}
```
In this way, you can inject various token and didDocument combinations into the `verify` function to unit test. Does this make sense?